include/circt/Conversion/CFToHandshake.h

```diff
@@ -162,30 +162,37 @@ template <typename T, typename... TArgs, typename... TArgs2>
 LogicalResult runPartialLowering(
     T &instance,
     LogicalResult (T::*memberFunc)(ConversionPatternRewriter &, TArgs2...),
-    TArgs &...args) {
+    TArgs &... args) {
   return partiallyLowerRegion(
       [&](Region &, ConversionPatternRewriter &rewriter) -> LogicalResult {
         return (instance.*memberFunc)(rewriter, args...);
       },
       instance.getContext(), instance.getRegion());
 }
 
+/// Remove basic blocks inside the given region. This allows the result to be
+/// a valid graph region, since multi-basic block regions are not allowed to
+/// be graph regions currently.
+void removeBasicBlocks(Region &r);
+
 // Helper to check the validity of the dataflow conversion
 // Driver that applies the partial lowerings expressed in HandshakeLowering to
 // the region encapsulated in it. The region is assumed to have a terminator of
-// type TTerm. See HandshakeLowering for the different lowering steps.
-template <typename TTerm>
+// type TSrcTerm, and will replace it with TDstTerm. See HandshakeLowering for
+// the different lowering steps.
+template <typename TSrcTerm, typename TDstTerm>
```

Comment:
I guess you have a downstream usage with different terminators than `handshake::ReturnOp`?